src/main/java/core/dao/manufacturer/ManufacturerDaoJdbc.java

```diff
@@ -0,0 +1,130 @@
+import core.model.Manufacturer;
+import core.utils.ConnectionUtils;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoJdbc implements ManufacturerDao {
+    @Override
+    public Manufacturer add(Manufacturer manufacturer) {
+        String insert = "INSERT INTO manufacturers(name, country)"
+                        + " VALUES(?, ?);";
+        
+        try (Connection con = ConnectionUtils.getConnection()) {
+            PreparedStatement insertStatement = con.prepareStatement(insert,
+                    Statement.RETURN_GENERATED_KEYS);
+            insertStatement.setString(1, manufacturer.getName());
+            insertStatement.setString(2, manufacturer.getCountry());
+            insertStatement.executeUpdate();
+            
+            ResultSet resultSet = insertStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject("id", Long.class));
+            }
+            insertStatement.close();
+            resultSet.close();
+        } catch (SQLException e) {
+            throw new DataProcessingException(String
+                    .format("Failed to insert the %s to database", manufacturer), e);
+        }
+        return manufacturer;
+    }
+    
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String select = "SELECT id, name, country"
+                        + " FROM manufacturers"
+                        + " WHERE (id = ? AND deleted = false);";
+        try (Connection connection = ConnectionUtils.getConnection()) {
+            PreparedStatement getById = connection.prepareStatement(select);
+            getById.setLong(1, id);
+            ResultSet resultSet = getById.executeQuery();
+            getById.close();
+            if (resultSet.next()) {
+                return Optional.of(getManufacturerFromResultSet(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException(String
+                    .format("Failed to get manufacturer by id = %d", id), e);
+        }
+    }
+    
+    @Override
+    public List<Manufacturer> getAll() {
+        String select = "SELECT id, name, country"
+                        + " FROM manufacturers"
+                        + " WHERE deleted = false;";
+        List<Manufacturer> resultList = new ArrayList<>();
+        try (Connection con = ConnectionUtils.getConnection()) {
+            PreparedStatement getAll = con.prepareStatement(select);
+            ResultSet resultSet = getAll.executeQuery();
+            getAll.close();
+            while (resultSet.next()) {
+                resultList.add(getManufacturerFromResultSet(resultSet));
+            }
+            resultSet.close();
```

Comment:
it's a bit redundant to close result set. If statement is closed, resultSet will be closed as well 
https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/Statement.html#close()

Reply:
Removed